viewing_party/party.py

```diff
@@ -0,0 +1,201 @@
+
+#this is the function that creates a new movie
+def create_movie(title, genre, rating):
+
+    # initilialize variables
+    
+    movie_dict = {}
+
+    #if these values are true, return dictionary
+    if title and genre and rating: #insert how to denote truthy and falsy values
+        movie_dict['title'] = title
+        movie_dict['genre'] = genre
+        movie_dict['rating'] = rating
+    else:
+        return None
+    
+    return movie_dict
+
+
+# function that that adds movie to watched list
+def add_to_watched(user_data, movie):
+    #add each movie in the 'watched' field of user_data
+    user_data['watched'].append(movie)
+    # return user-data
+    return user_data
+
+
+#add movie to movies to watch
+def add_to_watchlist(user_data, movie):
+    # user_data has a key - 'watchlist'
+    # value (which is what we are going to make) is a list of dictionaries 
+    # repping movies user wants to watch
+    
+    
+    # add movie to 'watchlist' key
+    user_data['watchlist'].append(movie)
+
+
+    #return user_data
+    return user_data
+
+
+
+# function that adds a movie to watched if given 
+def watch_movie(user_data,title):
+
+    # if title is a movie in user's watchlist
+    for movie in user_data['watchlist']:
+        
+        if title == movie['title']:
+            
+            user_data['watchlist'].remove(movie)
+            user_data['watched'].append(movie)
+    
+    return user_data
+
+
+# wave 2
+# function that takes the average rating of movies in watched
+def get_watched_avg_rating(user_data):
+    
+    rating_sum = 0
+    rating_list = []
+    
+    # get th ratings out of user data and put it in a list
+
+    for movie in user_data['watched']:
+        rating_sum += movie['rating']
+        rating_list.append(movie['rating'])
+    if len(rating_list) == 0:
+        avg_rating = 0
+    else:
+        avg_rating =rating_sum/len(rating_list)
+
+    
+    
+    return avg_rating
+
+
+
+# gets the genre from the list of movies that was watched the most
+def get_most_watched_genre(user_data):
+    genre_list = []
+    # user data is a dictionary w a watched list of movie dictionaries. in each element is a key genre and the value is a string
+    # see which genre appears the most and return it
+
+    # get to genres and add them to a list
+
+    for movie in user_data['watched']:
+        genre_list.append(movie['genre'])
+    
+    if len(user_data['watched']) == 0:
+        most_watched_genre = None
+    else:
+        most_watched_genre = max(set(genre_list), key = genre_list.count)
+
+    return most_watched_genre
+
+# wave 3
+# return a list of movies that the user has watched, but none of their friends have watched
+def get_unique_watched(user_data):
+    unique_user_watch_list = []
+    user_watched_list = []
+    friend_watched_list = []
+    for user_movie in user_data['watched']:
+        user_watched_list.append(user_movie)
```

Comment:
`user_data['watched']` is already a list of dictionaries so this loop is redundant. 

You could just assign `user_watched_list` to `user_data['watched']`.

```suggestion
    user_watched_list = user_data['watched']
    friend_watched_list = []
```